### PR TITLE
Use the latest official ECJ 3.14.0

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

Use the latest official ECJ 3.14.0
 
## Types of changes

- [X] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
The old ECJ reference was bogus (see official confirmation: https://bugs.eclipse.org/bugs/show_bug.cgi?id=499019#c19) and didn't work properly with JDK9

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -36,7 +36,7 @@ configurations {
 }
 
 dependencies {
-    ecj 'org.eclipse.jdt.core.compiler:ecj:4.6.1'
+    ecj 'org.eclipse.jdt:ecj:3.14.0'
```
[the remark this answers is not in the document]
Comment (valfirst) [reply build.gradle]: It seems, there are no other options. 
In general, I'm not aware of historical reasons of ECJ usage, in theory simple `javac` can work as well.

**Comment by mykola-mokhnach** [comment]
It looks like Travis is not very happy: https://travis-ci.org/appium/java-client/jobs/430697234

File: build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply build.gradle]: javac cannot compile the solution and https://github.com/SeleniumHQ/selenium/pull/863 is the main reason why it cannot

**Comment by valfirst** [comment]
It looks like some travis issue, similar to this one: https://github.com/travis-ci/travis-ci/issues/3795.

https://travis-ci.org/appium/java-client/jobs/430697234#L817:
```
attempting to download cache archive
fetching PR.1024/cache-linux-trusty-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--jdk-oraclejdk8.tgz
fetching PR.1024/cache--jdk-oraclejdk8.tgz
fetching master/cache-linux-trusty-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--jdk-oraclejdk8.tgz
running `casher fetch` took longer than 180 seconds and has been aborted.
You can extend the timeout with `cache.timeout`. See https://docs.travis-ci.com/user/caching/#Setting-the-timeout for details

adding /home/travis/.gradle/caches to cache
creating directory /home/travis/.gradle/caches
adding /home/travis/.gradle/wrapper to cache
creating directory /home/travis/.gradle/wrapper
/home/travis/.gradle/wrapper/dists/gradle-4.10.1-all/455itskqi2qtf0v2sja68alqd/gradle-4.10.1-all.zip: Permission denied
```

I'll touch the commit to trigger a new build

**Pull request merged**